xdsl/dialects/arith.py

```diff
@@ -190,6 +191,29 @@ class SignlessIntegerBinaryOperation(IRDLOperation, abc.ABC):
 
     assembly_format = "$lhs `,` $rhs attr-dict `:` type($result)"
 
+    @staticmethod
+    def py_operation(lhs: int, rhs: int) -> int | None:
+        return None
+
+    @staticmethod
+    def is_right_zero(attr: AnyIntegerAttr) -> bool:
+        """
+        Returns True only when 'attr' is a right zero for the operation
+        https://en.wikipedia.org/wiki/Absorbing_element
+
+        Note that this depends on the operation and does *not* imply that
+        attr.value.data == 0
+        """
+        return False
+
+    @staticmethod
+    def is_right_unit(attr: AnyIntegerAttr) -> bool:
+        """
+        Return True only when 'attr' is a right unit/identity for the operation
+        https://en.wikipedia.org/wiki/Identity_element
```

Comment:
ooh, if there are wikipedia pages with those titles then maybe using those for function names is fair game